### PR TITLE
updated byovpc network config for GCP

### LINKED ISSUE #1393: add support for byo vpc based workspace creation on gcp
please add GCP support to https://github.com/databricks/terraform-provider-databricks/blob/master/docs/resources/mws_networks.md , byo vpc on GCP feature - https://docs.gcp.databricks.com/administration-guide/cloud-configurations/gcp/customer-managed-vpc.html

### PR DESCRIPTION
**Pull request by stikkireddy**
- ~[ ] create doc updates/guide updates~
- [x] test creating mws for aws with older version and then upgrade and see impact on previous state
- [x] unit tests

Attempts to resolve #1393 

byovpc example:

```hcl
variable "databricks_account_id" {
  default = "myaccount-id"
}

variable "google_project" {
  default = "my-project"
}

terraform {
  required_providers {
    google = {
      source = "hashicorp/google"
      version = "4.37.0"
    }
    databricks = {
      source = "databricks/databricks"
      version = "1.6.5" # compile and install it as a new version
    }

  }
}


provider "google" {
  project     = "my-project"
  region      = "us-central1"
  credentials = ".../somecredfile.json"
}

provider "databricks" {
  host = "https://accounts.gcp.databricks.com"
  google_credentials = ".../somecredfile.json"
}

resource "random_string" "suffix" {
  special = false
  upper   = false
  length  = 6
}

resource "google_compute_network" "dbx_private_vpc" {
  project                 = var.google_project
  name                    = "tf-network-${random_string.suffix.result}"
  auto_create_subnetworks = false
#  mtu                     = 1460
}

resource "google_compute_subnetwork" "network-with-private-secondary-ip-ranges" {
  name          = "test-dbx-${random_string.suffix.result}"
  ip_cidr_range = "10.0.0.0/16"
  region        = "us-central1"
  network       = google_compute_network.dbx_private_vpc.id
  secondary_ip_range {
    range_name    = "pods"
    ip_cidr_range = "10.1.0.0/16"
  }
  secondary_ip_range {
    range_name    = "svc"
    ip_cidr_range = "10.2.0.0/20"
  }
  private_ip_google_access = true
}

resource "google_compute_router" "router" {
  name    = "my-router"
  region  = google_compute_subnetwork.network-with-private-secondary-ip-ranges.region
  network = google_compute_network.dbx_private_vpc.id
}

resource "google_compute_router_nat" "nat" {
  name                               = "my-router-nat"
  router                             = google_compute_router.router.name
  region                             = google_compute_router.router.region
  nat_ip_allocate_option             = "AUTO_ONLY"
  source_subnetwork_ip_ranges_to_nat = "ALL_SUBNETWORKS_ALL_IP_RANGES"
}

resource "databricks_mws_networks" "this" {
  account_id     = var.databricks_account_id
  network_name = "test-demo-${random_string.suffix.result}"
  gcp_network_info {
    network_project_id = var.google_project
    vpc_id = google_compute_network.dbx_private_vpc.name
    subnet_id = google_compute_subnetwork.network-with-private-secondary-ip-ranges.name
    subnet_region = google_compute_subnetwork.network-with-private-secondary-ip-ranges.region
    pod_ip_range_name = "pods"
    service_ip_range_name = "svc"
  }
}

resource "databricks_mws_workspaces" "this" {
 account_id     = var.databricks_account_id
 workspace_name = "tf-demo-test-${random_string.suffix.result}"
 location       = google_compute_subnetwork.network-with-private-secondary-ip-ranges.region
   cloud_resource_bucket {
     gcp {
       project_id = var.google_project
     }
   }
  network {
    network_id = databricks_mws_networks.this.network_id
    gcp_common_network_config {
      gke_connectivity_type = "PRIVATE_NODE_PUBLIC_MASTER"
      gke_cluster_master_ip_range = "10.3.0.0/28"
    }
  }
  depends_on = [google_compute_router_nat.nat]
}
```

### CODE REVIEW

File: mws/resource_mws_workspaces.go [hunk 1]
```diff
@@ -55,20 +55,20 @@ type CloudResourceBucket struct {
 }
 
 type GCPManagedNetworkConfig struct {
-	SubnetCIDR               string `json:"subnet_cidr"`
-	GKEClusterPodIPRange     string `json:"gke_cluster_pod_ip_range"`
-	GKEClusterServiceIPRange string `json:"gke_cluster_service_ip_range"`
+	SubnetCIDR               string `json:"subnet_cidr"  tf:"force_new"`
```
Comment (nfx) [inline mws/resource_mws_workspaces.go:58]: nit: please do one space instead of two

**Review by alexott — COMMENTED** [review]
code looks good

**Comment by nfx** [comment]
Following up - is this issue still relevant?

**Comment by nfx** [comment]
Following up - is this issue still relevant?

**Comment by martinzuern** [comment]
This feature is highly relevant for me. Is there anything I can do to move this forward? @nfx @stikkireddy

**Comment by nfx** [comment]
@martinzuern @yanw2 we cannot merge this without an acceptance test

**Comment by yanw2** [comment]
@nfx - Thank you for your reply. Can you please advise ETC if possible? Thanks.

**Comment by nfx** [comment]
etc?

**Comment by yanw2** [comment]
Sorry, I mean "Estimated to Complete" :)

**Comment by mahmuzicamel** [comment]
Is this pull request still a work in progress? Do you have any infos for us when it is available to use, since it is a kind of big relevance for my client? Thank you

**Comment by martinzuern** [comment]
@nfx what would you need for said acceptance test? Anything I can do to help here?

File: mws/acceptance/mws_workspaces_test.go [hunk 2]
```diff
@@ -85,3 +85,60 @@ func TestGcpAccaWorkspaces(t *testing.T) {
 		},
 	})
 }
+
+func TestGcpAccByovpcWorkspaces(t *testing.T) {
```
Comment (nfx) [inline mws/acceptance/mws_workspaces_test.go:89]:
```suggestion
func TestAccGcpByovpcWorkspaces(t *testing.T) {
```

**Review by nfx — APPROVED** [review]

**Comment by yanw2** [comment]
Hi @nfx, great thanks for fixing this. Can you please advise when you can publish this feature as a new release? It looks like it isn't included in the v1.7.0 release. Thank you.

**Pull request merged**